orbit/pkg/luks/luks_linux.go

```diff
@@ -0,0 +1,221 @@
+//go:build linux
+
+package luks_runner
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"regexp"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/orbit/pkg/dialog"
+	"github.com/fleetdm/fleet/v4/orbit/pkg/lvm"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/rs/zerolog/log"
+	"github.com/siderolabs/go-blockdevice/v2/encryption"
+	"github.com/siderolabs/go-blockdevice/v2/encryption/luks"
+)
+
+const (
+	entryDialogTitle     = "Enter disk encryption passphrase"
+	entryDialogText      = "Passphrase:"
+	retryEntryDialogText = "Passphrase incorrect. Please try again."
+	infoFailedTitle      = "Encryption key escrow"
+	infoFailedText       = "Failed to escrow key. Please try again later."
+	infoSuccessTitle     = "Encryption key escrow"
+	infoSuccessText      = "Key escrowed successfully."
+	maxKeySlots          = 8
+)
+
+var ErrKeySlotFull = regexp.MustCompile(`Key slot \d+ is full`)
+
+func (lr *LuksRunner) Run(oc *fleet.OrbitConfig) error {
```

Comment:
acknowledging a lack of unit test coverage on this method.  Much manual testing was done here on a physical machine (could not encrypt a VM) interacting with zenity prompts and monitoring LUKS states with `cryptsetup`